### ISSUE TITLE
Failure on Windows

### ISSUE THREAD

**Issue by ndmitchell** [mpickering/hie-bios#110]
If I change ghcide to have a different Stack component, I get:

```
ghcide.exe: CradleError (ExitFailure 1) ["Failed to parse result of calling stack","'stty' is not recognized as an internal or external command,","operable program or batch 
file.","\ESC[0mghcide> initial-build-steps (lib + exe)\ESC[0m","\ESC[0mConfiguring GHCi with the following packages: ghcide\ESC[0m","\ESC[0mReceived ExitFailure 1 when running","Raw command: \"C:\\\\Users\\\\ndmit_000\\\\AppData\\\\Local\\\\hie-bios\\\\wrapper-0.3.0.exe\" --version","Standard error:","","wrapper-0.3.0.exe: getEnv: invalid argument (The operation completed successfully.)","\ESC[0m"]
```

My guess is that's because one of the environment variables `HIE_BIOS_*` is not being set. I have no idea how they are meant to be set.

**Comment by mpickering** [comment]
@jneira Can you try to reproduce this please?

**Comment by ndmitchell** [comment]
Reproduction for me was the ghcide repo, changing `hie.yaml` to `cradle: {stack: {component: "ghcide:lib"}}`

**Comment by mpickering** [comment]
I tested on linux and it works correctly. The variables are set in `readProcessWithOutputFile`.

**Comment by jneira** [comment]
I can reproduce it directly calling the wrapper:
```
D:\dev\ws\haskell>C:\Users\atrey\AppData\Local\hie-bios\wrapper-0.3.0.exe --version
wrapper-0.3.0.exe: HIE_BIOS_OUTPUT: getEnv: does not exist (no environment variable)

D:\dev\ws\haskell>set HIE_BIOS_OUTPUT=.

D:\dev\ws\haskell>C:\Users\atrey\AppData\Local\hie-bios\wrapper-0.3.0.exe --version
wrapper-0.3.0.exe: HIE_BIOS_GHC: getEnv: does not exist (no environment variable)

D:\dev\ws\haskell>set HIE_BIOS_GHC=.

D:\dev\ws\haskell>C:\Users\atrey\AppData\Local\hie-bios\wrapper-0.3.0.exe --version
wrapper-0.3.0.exe: HIE_BIOS_GHC_ARGS: getEnv: does not exist (no environment variable)

D:\dev\ws\haskell>set HIE_BIOS_GHC_ARGS=.
.....
```

I've installed hie-bios with ec13a98 and i had to use the wrapper cause using `hie-bios check` give me another error:
```
PS D:\dev\ws\haskell\stack-test> hie-bios check .\src\MyLib.hs
hie-bios.exe: Cannot decode byte '\xbd': Data.Text.Internal.Encoding.streamDecodeUtf8With: Invalid UTF-8 stream
```

**Comment by jneira** [comment]
I've tested the commit before #97 and it works:
```
PS D:\dev\ws\haskell\stack-test> D:\bin\hie-bios\0a5cdc\hie-bios.exe check .\src\MyLib.hs
[1 of 1] Compiling Main             ( C:\TEMP\wrapper-0.3.0.hs, C:\TEMP\wrapper-0.3.0.o )
Linking C:\Users\atrey\AppData\Local\hie-bios\wrapper-0.3.0.exe ...
src\MyLib.hs:4:12:• Variable not in scope: putStrLnX :: [Char] -> IO ()
• Perhaps you meant ‘putStrLn’ (imported from Prelude)
```

**Comment by mpickering** [comment]
Anyone know why environmental variables don't work on windows?

**Comment by mpickering** [comment]
Is stack modifying the environment somehow to unset these variables?

**Comment by fendor** [comment]
I dont think so, otherwise the $HIE_BIOS_OUTPUT env variable wouldnt have been read either.

**Comment by mpickering** [comment]
Perhaps there is no difference on windows between setting a variable and setting it to `""`.

**Comment by mpickering** [comment]
Can we use `lookupEnv` rather than `getEnv` to gracefully handle this case?

**Comment by fendor** [comment]
yeah, that could be.
Of course.

**Comment by fendor** [comment]
This issue ought to be fixed by #112
